README.md

```diff
@@ -2,31 +2,140 @@
+- **Condition/Predicate Coverage:** The tests cause each boolean sub-expression in a compound boolean expression
+  to evaluate to `true` and `false` at least once.
+  - For example, the expression `(isRaining || isSnowing)` requires a minimum of 4 tests that evaluate as shown:
+    - `(true || true)`
+    - `(true || false)`
+    - `(false || true)`
+    - `(false || false)`
 
-- [Resource Link 1](example.com)
-- [Resource Link 2](example.com)
+
+## Black Box Testing
+
+**Black box testing** is a type of testing that is performed without any
+knowledge of the inner workings of the software. This means that black box
+testing can be performed by those who are not necessarily developing the
+software or writing any of its code. Examples of black box testing include:
+
+- **Functional testing**: testing specific functional aspects of a system,
+  regardless of how many underlying components are involved in implementing that
+  functionality. Some examples of functional test could be:
+    - Checking to make sure the functionality and features on a website is working
+      correctly. This could include looking at the menus, images, and links.
+    - Regression testing is usually thought to be a subset of functional testing
+      in that it double-checks that everything is working properly after a
+      change has been made.
+- **Non-functional testing**: testing the "non-functional" aspects of a system.
+  This type of testing looks more at the quality of the system rather than if
+  its features work as expected. Some examples of non-functional testing include:
+    - Performance testing to ensure the system can perform in a responsive and
+      timely manner despite several concurrent user requests.
+    - Security testing to test how secure the system is in terms of protecting
+      itself from malicious attacks.
+
+Non-functional testing, such as performance and security testing, is out of
+scope for this section, so let's focus on functional testing.
+
+It may seem like there would be a lot of overlap between "functional testing"
+and "unit testing", but the difference between the two is profound:
+
+- Unit testing focuses on making sure the code works as the developer intended
+  it to work.
+- Functional testing focuses on making sure the software works as the users need
+  it to work.
+
+In other words, one focuses on "building the thing right" while the other
+focuses on "building the right thing".
+
+This is a profound distinction because it does not matter if the code runs
+perfectly every time if it does not actually solve the problems that its users
+need solved.
+
+Let's consider a simplified version of a real-life example. Your team has been
+asked to build a "bill splitter" calculator:
+
+- There will be a number of participants.
+- There will be a total for the bill.
+- You've been asked to write the method does the division.
```

Comment:
Typo from me: "You've been asked to write the method does the division" should be "You've been asked to write the method that does the division" or "performs the division"

Reply:
fixed